### PR TITLE
Use USWDS 3.4.0

### PR DESCRIPTION
**Pull request by thisisdano**
In addition to using USWDS 3.4.0 and updating the references and links, this PR adds some content and guidance changes related to USWDS 3.4.0:

- Adds Sam.gov implementations (https://github.com/uswds/uswds-site/pull/1714)
- Adds Screen Reader instructions to Date Picker guidance (https://github.com/uswds/uswds-site/pull/1695)
- Adds font-face update documentation (https://github.com/uswds/uswds-site/pull/2022)
- Updates 404 template screenshots (https://github.com/uswds/uswds-site/pull/1923)
- Updates Gender Identity and Sex patterns (https://github.com/uswds/uswds-site/pull/1994)

[Preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/release-3.4.0/) →

### CODE REVIEW

**Comment by thisisdano** [comment]
Note that there's a temporary(?) unrelated 500 error in an external link that's unrelated to the content of this PR.

**Comment by thisisdano** [comment]
That 500 error is fixed now.

File: _components/prose/guidance/implementation.md [hunk 1]
```diff
@@ -4,6 +4,6 @@
   - `p`
   - `ul` and `ol`, _plus_ child `li`
   - `table`, _plus_ child `thead`, `th`, `td`, and `caption`
-- You can change the measure of elements styled with `usa-prose` in `_uswds-theme-typography.scss` with the following variables. Each accepts a measure token:
+- Customize line-length by changing the value of `$theme-text-measure` and `$theme-lead-measure` in your [USWDS settings configuration]({{ site.baseurl }}/documentation/settings). Each setting accepts a [measure token](https://designsystem.digital.gov/design-tokens/typesetting/measure/).
```
Comment (mejiaj) [inline _components/prose/guidance/implementation.md:7]:
```suggestion
- Customize line-length by changing the value of `$theme-text-measure` and `$theme-lead-measure` in your [USWDS settings configuration]({{ site.baseurl }}/documentation/settings). Each setting accepts a [measure token]({{ site.baseurl }}/design-tokens/typesetting/measure/). 
```

**Review by mejiaj — COMMENTED** [review]
Tiny comment on the link to measure. Added as suggestion; should be good to go after.

**Review by mejiaj — APPROVED** [review]

**Pull request merged**